proposals/0000-swiftpm-script-support.md

```diff
@@ -0,0 +1,156 @@
+
+The `@package(...)`syntax will allow the same parameters as SwiftPM's [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel natural to developers familiar with Swift Package Manager manifest syntax.  
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+
+Specifying the version of the package is optional. If omitted, SwiftPM will resolve the latest version of the dependency.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above package description, the `SwiftToolsSupport` product does not match a target within the package description so it must be specified explicitly.
+
+```swift
+@package(url: "https://github.com/apple/swift-tools-support-core.git", .exact("0.0.1"), products: ["SwiftToolsSupport"])
+import TSCBasic 
+
+```
+
+#### Dependency version requirements
+
+The `@package(..)` annotation would also allow a developer to specify requirements to ensure your scripts are reproducible. The syntax will contain the same level of control and specificity as `PackageDependencyDescription.Requirement` which should feel like a natural to developers familiar with Swift Package Manager manifest syntax.  
+
+*Revision*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .revision("5fa313eae1ca127ad3c706e14c564399989cb1b1")) 
+import Yams 
+```
+
+*Branch*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .branch("releases/1.0")) 
+import Yams 
+```
+
+*Exact Version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", .exact("2.0.0")) 
+import Yams 
+```
+
+*Range of version*
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "1.0.0") 
+import Yams 
+```
+
+*Local dependency*
+Absolute path
+```swift
+@package(path: "/Users/username/Yams") 
+import Yams 
+```
+
+Relative path
+```swift
+@package(path: "dependencies/Yams") 
+import Yams 
+```
+
+### Integration with Swift Compiler 
+
+The `@package()` syntax will need to be added to the AST definition and compiler front-end will require changes to parse this syntax.
+
+### Integration with SwiftPM
+
+#### Execution
+Scripts that utilize the `@package()` attribute can be executed by `swift run` and managed by SwiftPM.
```

Comment:
We should re-state that they can also be run directly with `swift script.swift`, which would then forward it to SwiftPM.